test/functional/editor/completion_spec.lua

```diff
@@ -1351,4 +1351,18 @@ describe('completion', function()
       eq('completeopt option does not include popup', api.nvim_get_var('err_msg'))
     end)
   end)
+
+  it([[does not include buffer from non-focusable window for 'complete' "w"]], function()
+    local buf = api.nvim_create_buf(false, true)
+    local cfg = { focusable = false, relative = 'win', bufpos = { 1, 0 }, width = 1, height = 1 }
+    api.nvim_open_win(buf, false, cfg)
+    api.nvim_buf_set_lines(buf, 0, -1, false, { 'foo' })
+    feed('i<C-N>')
+    screen:expect([[
+      ^                                                            |
+      {4:f}{1:                                                           }|
+      {1:~                                                           }|*5
+      {5:-- Keyword completion (^N^P) }{9:Pattern not found}              |
+    ]])
```

Comment:
for "positive confirmation" in this test, would also be good to set `focusable=true` and then assert that the contents *are* shown.